### PR TITLE
chore: add Coder for Docker docs

### PR DESCRIPTION
**Pull request by khorne3**

### CODE REVIEW

File: setup/docker.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+1. Launch Docker Desktop.
+
+1. If you've previously installed Coder, run `rm -rf ~/.coder` in the terminal.
+
+1. In the terminal, run the following to download the resources you need,
+   include the images, and set up your Coder deployment:
+
+   ```console
+   docker run -it --rm -p 7080:7080 -v ~/.coder:/var/run/coder -v /var/run/docker.sock:/var/run/docker.sock -e ENVBOX_IMAGE=kylecarbs/test-image-2:latest kylecarbs/test-image:0.4.0
```
Comment (khorne3) [inline setup/docker.md:24]: @kylecarbs could you please provide the updated command when you have it? thanks!

Comment (kylecarbs) [reply setup/docker.md:24]: It'll be:
```
docker run --rm -it -p 7080:7080 -v /var/run/docker.sock:/var/run/docker.sock -v ~/.coder:/var/run/coder codercom/coder
```
It's not live yet, but I'll put it up later today.

**Comment by ntimo** [comment]
I have been testing this out locally a bit, a few questions that came into mind when setting this up where not covered in the docs pr:
- How can I provide the devurl url
- How can I provide my own postgres
- How can I provide a SSL cert for SSL

File: setup/docker.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+command to start your deployment:
+
+```console
+-e DEVURL_HOST=yourHost
+```
+
+For example:
+
+```console
+docker run --rm -it -p 7080:7080 -v /var/run/docker.sock:/var/run/docker.sock -v ~/.coder:/var/run/coder codercom/coder -e DEVURL_HOST=yourHost
```
Comment (bpmct) [inline setup/docker.md:74]:
```suggestion
docker run --rm -it -p 7080:7080 -v /var/run/docker.sock:/var/run/docker.sock -v ~/.coder:/var/run/coder codercom/coder -e DEVURL_HOST="*.mycompany.com"
```

File: setup/docker.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+multi-machine scaling.
+
+For organizations, we recommend one Docker host per team of 5-10 developers.
+
+## Known issues
+
+Currently, Coder for Docker does not support:
+
+- External PostgreSQL databases
+- The use of your own TLS certificates
```
Comment (bpmct) [inline setup/docker.md:90]:
```suggestion
- The use of your own TLS certificates

To use TLS with Coder for Docker, you'll need to run Coder behind a reverse proxy,
such as Caddy or nginx and terminate TLS there.

```

**Review by bpmct — APPROVED** [review]

**Pull request merged**